sycl/source/device_selector.cpp

```diff
@@ -18,41 +19,30 @@ namespace sycl {
 
 // Utility function to check if device is of the preferred SYCL_BE.
 static bool isDeviceOfPreferredSyclBe(const device &Device) {
-  // Taking the version information from the platform gives us more useful
-  // information than the driver_version of the device.
-  const platform Platform = Device.get_info<info::device::platform>();
-  const std::string PlatformVersion =
-      Platform.get_info<info::platform::version>();
+  if (Device.is_host())
+    return false;
 
   detail::pi::Backend PreferredBE = detail::pi::getPreferredBE();
```

Comment:
each time isDeviceOfPreferredSyclBe/getPreferredBE is called there will be re-reading of SYCL_BE from environment - waste of time